src/prefect/orion/orchestration/rules.py

```diff
@@ -569,9 +569,19 @@ async def __aenter__(self) -> OrchestrationContext:
         if await self.invalid():
             pass
         else:
-            entry_context = self.context.entry_context()
-            await self.before_transition(*entry_context)
-            self.context.rule_signature.append(str(self.__class__))
+            try:
+                entry_context = self.context.entry_context()
+                await self.before_transition(*entry_context)
+                self.context.rule_signature.append(str(self.__class__))
+            except Exception as before_transition_error:
+                await self.abort_transition(
+                    f"Aborting orchestration due to error in {self.__class__!r}: !{before_transition_error!r}"
+                )
+                logger.error(
```

Comment:
Let's log the traceback
```suggestion
                logger.exception(
```